include/timer.h

```diff
@@ -44,16 +47,31 @@ static inline int64_t GetTicks(void)
 	        .count();
 }
 
+static inline int64_t GetTicksUs(void)
+{
+	return std::chrono::duration_cast<std::chrono::microseconds>(
+	               std::chrono::steady_clock::now().time_since_epoch())
+	        .count();
```

Comment:
When previously using chrono, every now and then I'd come across some stack exchange posts by a gentleman named Howard Hinnant. Here are a couple:

https://stackoverflow.com/a/27161931

And 

https://stackoverflow.com/a/36751930

I've double backed and started doing more learning about chrono, and turns out Howard actually implemented chrono for the C++11, 14, and so on! So it's no wonder his answers are so helpful. 

Both of those links above don't do justice to how efficient and powerful-yet-safe chrono actually is: check out his talk: https://m.youtube.com/watch?v=P32hvk8b13M I found my interest growing as the talk progressed, and by the time he mentioned the TRS80 emulator.. I was fully bought in and blown away by chrono 👍

How does all this relate to the PR? I think this PR is a good 1:1 replacement for the existing DOSBox delay calls - and I think a follow-on PR is possible that will switch the API to operate on literals, would be a perfect progression.

Anyway.. when you have a spare 45 min or so, watch that video and see what you think! 


I watched a YouTube video